File: backend/src/constants/handover_status.py
# 交接单生命周期状态
# PENDING   交班人已发起，等待接班人确认；交班人可撤回
# ACCEPTED  接班人已确认，任务正式转移给接班人
# REVOKED   接班人确认前被交班人撤回，交接失效
HANDOVER_STATUS = ["PENDING", "ACCEPTED", "REVOKED"]

HANDOVER_STATUS_TEXT = {
    "PENDING": "待接班确认",
    "ACCEPTED": "已接班",
    "REVOKED": "已撤回",
}

# 交接事件类型：每个动作都留痕，记录操作人、时间与前后负责人
HANDOVER_EVENT_TYPES = ["CREATE", "ACCEPT", "REVOKE"]

File: frontend/src/constants/HandoverStatus.ts
export const HandoverStatus = ["PENDING", "ACCEPTED", "REVOKED"] as const;
export type HandoverStatus = (typeof HandoverStatus)[number];

export const HandoverStatusText: Record<HandoverStatus, string> = {
  PENDING: "待接班确认",
  ACCEPTED: "已接班",
  REVOKED: "已撤回"
};

// 交接单操作事件：发起 / 确认接班 / 撤回
export const HandoverEventType = ["CREATE", "ACCEPT", "REVOKE"] as const;
export type HandoverEventType = (typeof HandoverEventType)[number];

export const HandoverEventTypeText: Record<HandoverEventType, string> = {
  CREATE: "发起交接",
  ACCEPT: "确认接班",
  REVOKE: "交班人撤回"
};
